internal/expr/scan.go

```diff
@@ -148,28 +172,59 @@ func (re *ResultExpr) decodeValue(v reflect.Value) error {
 	return nil
 }
 
-func setValue(dest reflect.Value, fInfo field, val any) error {
+func setValue(dest reflect.Value, fInfo fielder, val any) error {
 	var isZero bool
 
 	v := reflect.ValueOf(val)
+	name := fInfo.Name()
 
-	if val == nil {
-		if fInfo.omitEmpty {
-			return nil
+	switch dest.Type().Kind() {
+	case reflect.Struct:
+		f, ok := fInfo.(field)
+		if !ok {
+			return fmt.Errorf("internal error: argument of type %#v has no member of type %#v", dest.Type(), f)
 		}
-		isZero = true
-		v = reflect.Zero(fInfo.typ)
-	}
 
-	if !isZero && v.Type() != fInfo.typ {
-		return fmt.Errorf("result of type %#v but field %#v is type %#v", v.Type().Name(), fInfo.name, fInfo.typ.Name())
-	}
-	f := dest.FieldByIndex(fInfo.index)
-	if !f.CanSet() {
-		return fmt.Errorf("cannot set field %#v. CanAddr=%v", fInfo.name, f.CanAddr())
+		if val == nil {
+			if f.omitEmpty {
+				return nil
+			}
+			isZero = true
+			v = reflect.Zero(f.typ)
+		}
+		if !isZero && v.Type() != f.typ {
+			return fmt.Errorf("result of type %#v but field %#v is type %#v", v.Type().Name(), name, f.typ.Name())
+		}
+		itsField := dest.FieldByIndex(f.index)
+		if !itsField.CanSet() {
+			return fmt.Errorf("cannot set field %#v. CanAddr=%v", name, itsField.CanAddr())
+		}
+		itsField.Set(v)
+		return nil
+	case reflect.Map:
+		m, ok := fInfo.(mapKey)
+		if !ok {
+			return fmt.Errorf("internal error: argument of type %#v has no member of type %#v", dest.Type(), m)
+		}
+
+		if val == nil {
```

Comment:
I don't think we need to worry about this case. If `val == nil` then it is because there was `NULL` stored in the database here, we can just set the map value to `nil` like it was any other value. We worry about this in the struct case becuase of the `omitEmpty` flag.